baiji/serialization/json.py

```diff
@@ -23,83 +24,140 @@ def _load(f, *args, **kwargs):
 def loads(*args, **kwargs):
     return json.loads(*args, **_load_args(kwargs))
 
-def _load_args(kwargs):
-    kwargs.update(object_hook=_json_decode)
-    return kwargs
+class MethodListCaller(object):
+    def register(self, method, index=-1):
+        if not hasattr(self, 'method_list'):
+            self.clear()
+        if index == -1:
+            index = len(self.method_list)
+        self.method_list.insert(index, method)
 
-def _dump_args(kwargs):
-    if kwargs.get('primitive', False):
-        kwargs.update(default=_json_encode_primitive)
+    def clear(self):
+        # be defensive if someone forgets to call super pylint: disable=attribute-defined-outside-init
+        self.method_list = []
+
+    def __call__(self, x):
```

Comment:
Elsewhere I have seen the pattern of placing actual computations in a separate method and making`__call__` simply relay to this other method. I kinda liked it.
It is not something I feel strongly about but worth considering.